project/SparkBuild.scala

```diff
@@ -765,16 +750,6 @@ object CopyDependencies {
 
 }
 
-object Java8TestSettings {
-  import BuildCommons._
-
-  lazy val settings = Seq(
-    javacJVMVersion := "1.8",
-    // Targeting Java 8 bytecode is only supported in Scala 2.11.4 and higher:
-    scalacJVMVersion := (if (System.getProperty("scala-2.10") == "true") "1.7" else "1.8")
```

Comment:
I think I know what I missed; these are the settings for the java 8 project, which you're removing, so they're not gonna make a difference.

But I wonder how the 2.10 compile works given that you're setting the scalac version to 1.8 in L237. Maybe I missed something else.